weather_data_analyzer.py

```diff
@@ -0,0 +1,198 @@
+import csv
+import sys
+from weather_record import WeatherRecord
+from report_types \
+    import AverageTempHumidReportType, HighLowTempHumidityReportType
+
+dummy_record_for_comparison = {
+    'PKT': "2022-08-16",
+    'Mean TemperatureC': None,
+    'Max TemperatureC': -sys.maxsize,
+    'Min TemperatureC': sys.maxsize,
+    ' Min Humidity': sys.maxsize,
+    'Max Humidity': -sys.maxsize,
+    ' Mean Humidity': None
+}
+
+
+class WeatherDataAnalyzer:
+    """
+    Receives the file paths and store their record in the list. Then perform
+    different calculations on that list
+    """
+
+    def __init__(self, filepaths) -> None:
+
+        self.file_records = []
+
+        for filepath in filepaths:
+            self.file_data_reader_and_add_to_file_records(filepath)
+
+    def file_data_reader_and_add_to_file_records(self, filepath):
+        """
+        Read the data of given file and append it to the list
+
+        :param filepath:
+        """
+
+        with open(filepath, 'r') as file:
+            records_list = csv.DictReader(file)
+
+            for row in records_list:
+                record = WeatherRecord(row)
+                self.file_records.append(record)
+
```

Comment:
its fine but the next step can be of using pandas to read and clean data from source files.

Reply:
Thanks. I will try with pandas.